pkg/kubevirt/client.go

```diff
@@ -325,7 +325,7 @@ func (c *client) getSnapshotClassNameFromVolumeClaimName(ctx context.Context, na
 	}
 	if storageClassName == "" && !c.storageClassEnforcement.AllowDefault {
 		return "", fmt.Errorf("unable to determine volume snapshot class name for snapshot creation, and default not allowed")
-	} else if storageClassName != "" && !(util.Contains(c.storageClassEnforcement.AllowList, storageClassName) || c.storageClassEnforcement.AllowAll) {
+	} else if storageClassName != "" && !(util.Contains(c.storageClassEnforcement.AllowList, storageClassName) || c.storageClassEnforcement.AllowAll) && !c.storageClassEnforcement.AllowDefault {
```

Comment:
I'm not sure this logic is entirely complete.

Here's what I see with human readable comments.

```
if storageClassName == "" && !c.storageClassEnforcement.AllowDefault {
// reject, default not allowed
} else if storageClassName != "" && !(util.Contains(c.storageClassEnforcement.AllowList, storageClassName) || c.storageClassEnforcement.AllowAll) && !c.storageClassEnforcement.AllowDefault {
// reject, storage class is not in the allow list and default is not allowed.
}

// pass unconditionally if storageClassName != "" and default is allowed
```

Doesn't this mean that if a storageClass is found, is not in the allow list, but default is allowed, then the storageClass will pass this check? 

basically if AllowDefault is true, and a storageClass is found, then the allow list is ignored

Reply:
Right, but this function looks up the name of the matching snapshot class. This pass through will then return blank, which will leave the volume snapshot class in the snapshot empty and thus use the default snapshot class if it is set.